### PR TITLE
Expect max score in the index definition

### PR DESCRIPTION
**Pull request by miccolis**
### Context

As described in #853 we package frequency info, but don't actually use it data at query time. This PR removes it and demands that the one value we do use - `maxscore` is written to the index metadata. This will be a major breaking change.

The intent is that this is a temporary situation, we aim to make frequency available again at query time and use it. This sets the stage for that work. 

A serious downside is that until we actually bring frequency information back to the index itself Carmen doesn't write a max score to the index definition, that responsibility falls to the application/user which manages index construction.

### Next Steps

- [x] Remove some code which looped over `['grid', 'freq']` arrays during index construction.
- [x] Downstream testing.

### Draft changelog entry

```
28.0.0

* No longer generate or read the frequency.rocksdb file
* Demand that the `maxscore` is in index metadata
* Remove out-of-date `bin/tokenize.js` script
```

cc @mapbox/search

### CODE REVIEW

File: test/unit/indexer/index.test.js [hunk 1]
```diff
@@ -105,8 +105,7 @@ test('index.update -- error', (t) => {
             }
         }], { zoom: 6 }, (err) => {
             q.ifError(err);
-            q.deepEqual(conf.to._geocoder.freq.get('__COUNT__'), [2]);
-            q.deepEqual(conf.to._geocoder.freq.get('__MAX__'), [10]);
+            q.deepEqual(conf.to._geocoder.grid.list(),[['main st', [0]]]);
```
Comment (miccolis) [inline test/unit/indexer/index.test.js]: This doesn't look right to me and I'm not quite sure what the intent of the original test was. Need another look.

Comment (apendleton) [reply test/unit/indexer/index.test.js]: The original intent was to verify that the `__COUNT__` and `__MAX__` values were right in the freq table -- that the number of words we've seen is 2 (I think that's what `__COUNT__` was?), and the highest score is 10. What's happening now is a list in the grid table instead -- that we have one phrase, included for language set `0` (so, default text only). That's as expected from the test, but I'm not sure it has much to do with what the test was doing before.

Comment (miccolis) [reply test/unit/indexer/index.test.js]: Right, count it the terms, not entries. Great. So, should I just :hocho: this test?

**Comment by miccolis** [comment]
@apendleton I think I addressed the test side changes you requested. Let me know.

**Pull request merged**